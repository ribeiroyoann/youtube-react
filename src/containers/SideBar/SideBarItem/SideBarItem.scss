@import "../../../styles/shared.scss";

.sidebar-item {
  background: #f5f5f5;
  span {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    i.icon {
      margin-right: 20px;
      color: $grey;
    }
  }

  &.highlight-item {
    span {
      font-weight: 600;
    }

    i.icon {
      color: $red;
    }
  }
}

.sidebar-item-alignment-container {
  display: flex;
  align-items: center;
}
