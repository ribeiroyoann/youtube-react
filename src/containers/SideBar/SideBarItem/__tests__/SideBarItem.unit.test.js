import React from "react";
import { shallow } from "enzyme";
import { SideBarItem } from "../SideBarItem";

describe("SideBarItem", () => {
  test("renders empty SideBarItem", () => {
    const wrapper = shallow(<SideBarItem />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders highlighted SideBarItem that navigates to /feed/trending", () => {
    const wrapper = shallow(
      <SideBarItem highlighted icon="fire" label="Trending" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("renders SideBarItem to history", () => {
    const wrapper = shallow(<SideBarItem icon="history" label="History" />);
    expect(wrapper).toMatchSnapshot();
  });
});
