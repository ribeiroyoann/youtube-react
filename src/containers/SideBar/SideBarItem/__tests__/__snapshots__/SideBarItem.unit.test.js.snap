// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem renders SideBarItem to history 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="history"
        size="large"
      />
    </span>
    <span>
      History
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem renders empty SideBarItem 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`SideBarItem renders highlighted SideBarItem that navigates to /feed/trending 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;
